src/main/java/fruitshop/strategy/ReturnOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.set(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded");
+        }
```

Comment:
The check for a non-negative balance makes sense, but since this is a Return operation, it's not expected to have a situation where the balance could become negative. The check might be unnecessary here. If you still decide to keep this check, the error message should be more informative and specific to the Return operation context.